### ISSUE TITLE
Add strength argument to random number generation calls

### ISSUE THREAD

**Issue by paulidale** [openssl/openssl#15512]
There are a number of new calls introduced relating to library contexts and random number generation:

- `BN_rand_ex()`
- `BN_priv_rand_ex()`
- `BN_rand_range_ex()`
- `BN_priv_rand_range_ex()`
- `RAND_bytes_ex()`
- `RAND_priv_bytes_ex()`

It would make sense to also introduce an extra argument to these calls `unsigned int strength` to includes the security strength required of the generation.

The underlying EVP_RAND_CTX generate calls support this, so it's a matter of plumbing the argument through.  This part is straightforward and would need to be done before beta.

All calls to these functions from within libcrypto and libssl would need to be updated, however it wouldn't be necessary for a value other than zero to be passed for the moment.  That is, the full enforcement of security strengths could be postponed.

**Comment by slontis** [comment]
@paulidale This should be closed now?

**Comment by paulidale** [comment]
I expect we'll need a decision as to enforcement or not.
It doesn't need to be tracked here but might as well be.

**Comment by paulidale** [comment]
Closing this.  Decision in #15502.
